### PR TITLE
Install the common test executables to libexec

### PR DESCRIPTION
**Pull request by scpeters**
# 🎉 New feature

Part of https://github.com/gazebosim/gz-physics/issues/50#issuecomment-747095048

## Summary

This installs the `COMMON_TEST_*` executables to a subfolder of `libexec` following the pattern used by the standalone executables added in https://github.com/gazebosim/gz-transport/pull/216. This will allow these tests to be used by external implementers of `gz-physics` plugins. The `test_lib_loader` library is changed to a static library as well since these executables link against it.

Remaining tasks:

* [ ] Install `resources` folder used by some tests
* [ ] Install `worlds` folder used by some tests
* [ ] Provide cmake helper for adding tests to downstream projects using these executables

## Test it

* Build and install this branch
* Run `<prefix>/libexec/gz/physics6/COMMON_TEST_construct_empty_world <prefix>/lib/libgz-physics6-dartsim-plugin.<suffix>`

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by scpeters** [comment]
I've updated `TestLibLoader.hh` to be header-only in https://github.com/gazebosim/gz-physics/pull/458/commits/57ec5eb7c8e3e5b96ffc5b4cb8eb21dd7027dac9, hoping that this fixes our windows CI issues

**Comment by scpeters** [comment]
this is ready for review

**Review by ahcorde — APPROVED** [review]

File: test/common_test/TestLibLoader.hh [hunk 1]
```diff
@@ -66,5 +74,12 @@ namespace physics
     }
     return "";
   }
+
+  private: static std::string libToTest;
+};
 }
 }
+
+std::string gz::physics::TestLibLoader::libToTest = std::string("");
```
Comment (azeey) [inline test/common_test/TestLibLoader.hh:83]: Defining a static variable in a header can be [problematic](https://stackoverflow.com/questions/18860895/how-to-initialize-static-members-in-the-header). I recommend putting the variable in a private function and make it a singleton.

```
private: static std::string& TestLibLoader::LibToTest() 
{
  static std::string libToTest = "";
  return libToTest;
}
```

Comment (scpeters) [reply test/common_test/TestLibLoader.hh]: how does https://github.com/gazebosim/gz-physics/pull/458/commits/18f950c48f4a5a3171ce026e54cc920e06523d6f  look?

**Review by azeey — APPROVED** [review]

**Pull request merged**